### DISCUSSION TITLE
cannot reuse already awaited aclose()/athrow()

### DISCUSSION THREAD

**Discussion by alinemone** [encode/httpx#3005] [General]
On Windows systems, when I use this method

async def get_api(cls):
        a = await httpx.AsyncClient().get('https://api.publicapis.org/entries')
I get the following error

    "type": "RuntimeError",
    "message": "cannot reuse already awaited aclose()/athrow()",
    "trace": [
        {
            "file": "E:\\Programs\\order\\venv\\lib\\site-packages\\httpx\\_client.py",
            "name": "_send_handling_auth",
            "line number": 1662,
            "line": "await auth_flow.aclose()"
        },
        
        
I commented line 1662 in the _client.py file
 await auth_flow.aclose()
        
and the problem was solved, but I don't know if others have encountered the problem. This part has a problem on Windows

**Comment by lovelydinosaur** [comment]
Curious, we've not had this reported before from our windows userbase. Anyone able to replicate?

@alinemone Are you able to try using `trio` instead of `asyncio` and let us know if the issue presents there too?

**Reply by alinemone** [reply]
```
import httpx
import trio


async def ping():
    _client = httpx.AsyncClient()
    _client.base_url = 'http://auth.xxxx.com'
    response = await _client.get('ping')
    if response.status_code == 200:
        print("Request successful")
        res = response.json()
        print(res)


trio.run(ping)
```
I tested this way, but the result was the same as the previous error

I also tested it with the asyncio package, and it showed the following error

**RuntimeError: Event loop is closed**

**Reply by zanieb** [reply]
It looks like you're not using `async with` to properly context manage your client, can you try that please?
